python/ray/worker.py

```diff
@@ -1523,6 +1523,7 @@ def process_task(task): # wrapping these lines in a function should cause the lo
                                         "function_name": function_name})
 
   check_main_thread()
+  worker.photon_client.register_pid()
```

Comment:
I think it is more appropriate for this to be part of `photon_connect`. I needed something similar for the actor PR (to pass an "actor ID" from the worker to the local scheduler), so we should probably use the same mechanism.

I did it like this https://github.com/ray-project/ray/pull/242/commits/5b2275077060b4097288d3c637cad0f300e994a2#diff-2a906576526ef8fb8a2de97b6e0bc3ceR7

What do you think about doing it that way? (Your PR will be ready first)

NOTE: the local scheduler should never terminate a driver, so if we do it in `photon_connect`, we will need the option of passing `-1` or something to indicate that it is a driver. (but this will be useful anyway because the local scheduler really ought to know which clients are drivers (for computing resource capacity).